File: scripts/check-site.mjs
import { readdir, readFile, stat } from "node:fs/promises";
import { join } from "node:path";

const root = new URL("..", import.meta.url).pathname;
const required = ["index.html", "daily/index.html", "daily/2026-06-29/index.html", "daily/2026-06-29/en/index.html", "assets/styles.css", "assets/app.js", "sitemap.xml"];
const missing = [];

for (const file of required) {
  try {
    await stat(join(root, file));
  } catch {
    missing.push(file);
  }
}

const htmlFiles = [];
async function walk(dir) {
  const entries = await readdir(dir, { withFileTypes: true });
  for (const entry of entries) {
    const full = join(dir, entry.name);
    if (entry.isDirectory()) {
      await walk(full);
    } else if (entry.name.endsWith(".html")) {
      htmlFiles.push(full);
    }
  }
}

await walk(root);

const broken = [];
for (const file of htmlFiles) {
  const html = await readFile(file, "utf8");
  if (!html.includes("VLYQB1HXUW")) broken.push(file);
  if (!html.includes("/assets/styles.css?v=20260630-gate-blue")) broken.push(file);
}

if (missing.length || broken.length) {
  console.error({ missing, broken });
  process.exit(1);
}

console.log(`Checked ${htmlFiles.length} HTML files successfully.`);

File: scripts/check-site.py
#!/usr/bin/env python3
from pathlib import Path

required = [
    "index.html",
    "zh-hant/index.html",
    "en/index.html",
    "ru/index.html",
    "daily/index.html",
    "zh-hant/daily/index.html",
    "en/daily/index.html",
    "ru/daily/index.html",
    "daily/2026-07-02/index.html",
    "zh-hant/daily/2026-07-02/index.html",
    "en/daily/2026-07-02/index.html",
    "ru/daily/2026-07-02/index.html",
    "daily/2026-07-01/index.html",
    "zh-hant/daily/2026-07-01/index.html",
    "en/daily/2026-07-01/index.html",
    "ru/daily/2026-07-01/index.html",
    "daily/2026-06-29/index.html",
    "zh-hant/daily/2026-06-29/index.html",
    "en/daily/2026-06-29/index.html",
    "ru/daily/2026-06-29/index.html",
    "assets/styles.css",
    "assets/app.js",
    "sitemap.xml",
]

missing = [path for path in required if not Path(path).exists()]
htmls = list(Path(".").glob("**/*.html"))
broken = []

for path in htmls:
    text = path.read_text(encoding="utf-8", errors="ignore")
    checks = [
        "GateAffiliate",
        "VLYQB1HXUW",
        "/assets/styles.css?v=20260701-tg",
        'hreflang="zh-CN"',
        'hreflang="zh-Hant"',
        'hreflang="en"',
        'hreflang="ru"',
    ]
    if any(check not in text for check in checks):
        broken.append(str(path))

sitemap = Path("sitemap.xml").read_text(encoding="utf-8")
expected_urls = 76
if sitemap.count("<url>") != expected_urls:
    broken.append("sitemap.xml:url-count")

if missing or broken:
    print({"missing": missing, "broken": broken})
    raise SystemExit(1)

print(f"Checked {len(htmls)} HTML files and {expected_urls} sitemap URLs successfully.")
